### PR TITLE
docs: clarify offline license options

### PR DESCRIPTION
**Pull request by Meschreiber**
Clarify that the `--license <license_path>` option is only available when running the router binary, not for use with `rover dev`

### CODE REVIEW

**Review by abernix — APPROVED** [review]

File: docs/source/enterprise-features.mdx [hunk 1]
```diff
@@ -104,7 +104,12 @@ Follow these steps to configure an Apollo Router to use an offline Enterprise li
     1. [`APOLLO_ROUTER_LICENSE_PATH`](./configuration/overview/#--license) environment variable, containing an absolute or relative path to an offline license file 
     1. [`APOLLO_ROUTER_LICENSE`](./configuration/overview/#--license) environment variable, containing the stringified contents of an offline license file
 
-    The router checks the CLI option and environment variables in the listed order, then it uses the value of the first option or variable that is set.
+    <Note>
+
+    - The router checks the CLI option and environment variables in the listed order, then it uses the value of the first option or variable that is set.
```
Comment (lrlna) [inline docs/source/enterprise-features.mdx:109]: maybe "provided order" is clearer?
```suggestion
    - The router checks the CLI option and environment variables in the provided order, then it uses the value of the first option or variable that is set.
```

**Review by lrlna — APPROVED** [review]
Looks good @Meschreiber!

**Comment by lrlna** [comment]
@Meschreiber does this have to target `main`, not `dev`? i think that's why snapshots are failing

**Comment by Meschreiber** [comment]
@lrlna , ah interesting. Was hoping to target main so this gets published right away w/o waiting for the next release cycle. I know we've done this in the past for certain doc updates. 🤔

**Comment by lrlna** [comment]
@Meschreiber I just pushed a fix with updated snapshots. I think all the doc PRs you have open should pass after this one is merged.

**Pull request merged**